main/lsp/requests/code_action.cc

```diff
@@ -29,8 +29,229 @@ vector<unique_ptr<TextDocumentEdit>> getEdits(const LSPConfiguration &config, co
+            if (maybeMethod.has_value()) {
+                return maybeMethod.value();
+            }
+        }
+    } else if (auto klass = ast::cast_tree<ast::ClassDef>(tree)) {
+        for (auto &subtree : klass->rhs) {
+            auto maybeMethod = findMethodTree(subtree, method);
+            if (maybeMethod.has_value()) {
+                return maybeMethod.value();
+            }
+        }
+    } else if (auto methodDef = ast::cast_tree<ast::MethodDef>(tree)) {
+        if (methodDef->symbol == method.asMethodRef()) {
+            return methodDef;
+        }
+    }
+    return std::nullopt;
+}
+
+unique_ptr<string> copyMethodSource(const core::GlobalState &gs, const core::LocOffsets &sigLoc,
+                                    const core::LocOffsets &methodLoc, const core::FileRef &fref) {
+    auto cutSource = [&](core::LocOffsets loc) {
+        return fref.data(gs).source().substr(loc.beginPos(), loc.endPos() - loc.beginPos());
+    };
+
+    auto sigSource = cutSource(sigLoc);
+    auto methodSource = cutSource(methodLoc);
+    return make_unique<string>(absl::StrCat(sigSource, "\n  ", methodSource));
+}
+
+std::optional<std::pair<core::LocOffsets, core::LocOffsets>> methodLocs(const core::GlobalState &gs,
+                                                                        const ast::ExpressionPtr &rootTree,
+                                                                        const core::SymbolRef &method,
+                                                                        const core::FileRef &fref) {
+    auto maybeTree = findMethodTree(rootTree, method.asMethodRef());
+    if (!maybeTree.has_value()) {
+        return std::nullopt;
+    }
+    auto methodLoc = maybeTree.value()->loc;
+
+    auto maybeSig = sorbet::sig_finder::findSignature(gs, method);
+    if (!maybeSig.has_value()) {
+        return std::nullopt;
+    }
+    core::LocOffsets sigLoc = {maybeSig->sig.beginPos(), maybeSig->body.endPos()};
+
+    return make_pair(sigLoc, methodLoc);
+}
+
+// Turns ruby_function_name__ to RubyFunctionName,
+// so we can use it in a new module name
+string snakeToCamelCase(string_view name) {
+    string res;
+    const auto originalSize = name.size();
+    res.reserve(originalSize);
+    bool shouldCapitalize = true;
+    for (int i = 0; i < originalSize; i++) {
+        if (name.at(i) == '_') {
+            shouldCapitalize = true;
+            continue;
+        }
+        res += shouldCapitalize ? toupper(name[i]) : name[i];
+        shouldCapitalize = false;
+    }
+    return res;
+}
+
+optional<string> getNewModuleName(const core::GlobalState &gs, const core::NameRef name) {
+    const auto moduleName = absl::StrCat(snakeToCamelCase(name.show(gs)), "Module");
+    const auto nameNotExists = [&](auto moduleName) {
+        return gs.lookupNameUTF8(moduleName) == core::NameRef::noName();
+    };
+
+    // if there are no names like that, return the new module name
```

Comment:
ooc, was it intentional that you're doing this on names, not on symbols?

Doing this on names is ~fine, but it does mean there are some weird things, like

```ruby
puts(:GreetingModule)
```

Sorbet entered a name for `GreetingModule`, so it's going to cause the fallback logic to run, even if there's no module like `module GreetingModule`.

If it's intentional, can you write a test like what I've shown here and update this comment to explain the reasoning?

If you decide to update this to use symbols instead of names, I don't mind if you leave the comment and tests as they are.